logstash-core/src/main/java/org/logstash/secret/SecretIdentifier.java

```diff
@@ -60,15 +69,19 @@ public static SecretIdentifier fromExternalForm(String urn) {
     /**
      * Minor validation and downcases the parts
      *
-     * @param part     The part of the URN to validate
+     * @param key      The key, a part of the URN to validate.
      * @param partName The name of the part used for logging.
      * @return The validated and transformed part.
      */
-    private static String validateWithTransform(String part, String partName) {
-        if (part == null || part.isEmpty()) {
+    private static String validateWithTransform(final String key, final String partName) {
+        if (key == null || key.isEmpty() || Strings.isBlank(key)) {
             throw new IllegalArgumentException(String.format("%s may not be null or empty", partName));
         }
-        return part.toLowerCase(Locale.US);
+
+        if (!KEY_PATTERN.matcher(key).matches()) {
+            logger.warn(String.format("Invalid %s key appeared in keystore. Please remove it as it cannot be used in the pipelines", key));
```

Comment:
nit: backtick-quote the user input

```suggestion
            logger.warn(String.format("Invalid `%s` key appeared in keystore. Please remove it as it cannot be used in the pipelines", key));
```

Reply:
If we take my other suggestion to have the pattern and its description available from `ConfigVariableExpander`, this will also benefit from including the description of the pattern.